UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -705,12 +673,13 @@ BodyPartType bodyPartAim
 		{
 			LastDamagedBy = damagedBy;
 
-			foreach (var bodyPartContainer in RootBodyPartContainers)
+			foreach (var bodyPart in BodyPartList)
 			{
-				if (bodyPartContainer.BodyPartType == bodyPartAim)
+				if (bodyPart.BodyPartType == bodyPartAim)
 				{
 					//Assuming is only going to be one otherwise damage will be duplicated across them
-					bodyPartContainer.TakeDamage(damagedBy, damage, attackType, damageType, armorPenetration);
```

Comment:
so,  Since this was expecting root body containers, 
the extra logic for having multiple body parts in an area would be handled by them

so, this has to be changed to take account of how many body parts are there, 
so for Bomb , Fire , Rad it splits the damage over all body parts in target area

For everything else it picks a random body part and applies the damage to that

Reply:
solved on last commit